File: org/kawaiilang/scr/Interpreter/Variable.java
package org.kawaiilang;

class Variable {

  private Token type;
  private Object value;
  private int scopeDepth;

  public Variable(Token type, Object value, int scopeDepth) {
    this.type = type;
    this.value = value;
    this.scopeDepth = scopeDepth;
  }

  public Variable(Token type, Object value) {
    this.type = type;
    this.value = value;
  }

  public Token getType() {
    return type;
  }

  public Object getValue() {
    return value;
  }

  public int scopeDepth() {
    return scopeDepth;
  }

}

File: KawaiiSwift/KawaiiSwift/Variable.m
//
//  Variable.m
//  KawaiiSwift
//
//  Created by Albert and Mike on 2021-01-09.
//

#import <Foundation/Foundation.h>
#import "Token.m"
#import "Variable.h"

@implementation Variable

- (Variable*) init:(Token*) type value:(NSObject*) value {
    self = [super init];
    if (self) {
        self->type = type;
        self->value = value;
    }
    return self;
}

- (Token*) getType {
    return type;
}

- (NSObject*) getValue {
    return value;
}

@end
